input.cc

```diff
@@ -233,18 +233,26 @@ static void read_phixs_data_table(FILE *phixsdata, const int nphixspoints_inputt
 }
 
 static void read_phixs_data(int phixs_file_version) {
-  globals::nbfcontinua_ground = 0;
-  globals::nbfcontinua = 0;
   long mem_usage_phixs = 0;
 
   printout("readin phixs data from %s\n", phixsdata_filenames[phixs_file_version]);
 
   FILE *phixsdata = fopen_required(phixsdata_filenames[phixs_file_version], "r");
 
-  if (phixs_file_version == 1) {
+  const bool phixs_v1_exists = std::ifstream(phixsdata_filenames[1]).good();
+  const bool phixs_v2_exists = std::ifstream(phixsdata_filenames[2]).good();
+  if (phixs_file_version == 1 && phixs_v2_exists) {
+    printout(
+        "using NPHIXSPOINTS = %d and NPHIXSNUINCREMENT = %lg from phixsdata_v2.txt to interpolate "
+        "phixsdata.txt data\n",
+        globals::NPHIXSPOINTS, globals::NPHIXSNUINCREMENT);
+    last_phixs_nuovernuedge = (1.0 + globals::NPHIXSNUINCREMENT * (globals::NPHIXSPOINTS - 1));
+  } else if (phixs_file_version == 1) {
     globals::NPHIXSPOINTS = 100;
     globals::NPHIXSNUINCREMENT = .1;
-    last_phixs_nuovernuedge = 10;
+    last_phixs_nuovernuedge = 10;  // this matches classic
+    printout("using NPHIXSPOINTS = %d and NPHIXSNUINCREMENT = %lg set in input.cc to interpolate phixsdata.txt data\n",
+             globals::NPHIXSPOINTS, globals::NPHIXSNUINCREMENT);
```

Comment:
Accessing the file system every time a cross section is needed is going to cause serious performance issues. Replace the global int phixs_file_version with bool phixs_file_version_exists[] and just set them once in input.cc.

Reply:
I'm not sure I understand sorry, how does information on which phixsdata files exist from phixs_file_version_exists[] help determine which phixsdata file is being read on a given iteration of read_phixs_data() if both phixsdata files exist?

Reply:
Continue passing the phixs_file_version argument to read_phixs_data(). Just don't make it a global variable.

Reply:
phixs_file_version is used in the photoionization_crosssection_fromtable in atomic.cc. Should I retain a seperate global variable for phixs_file_version e.g. phixs_file_version_glob which I just use for this case?

Reply:
I suggested replacing that global variable with a new global bool phixs_file_version_exists[]. What number would phixs_file_version contain if both files are being used?

Reply:
if (phixs_file_version == 1) {
    // classic mode: no interpolation
    if (nu == nu_edge) {
      sigma_bf = photoion_xs[0];
    } else if (nu <= nu_edge * (1 + globals::NPHIXSNUINCREMENT * globals::NPHIXSPOINTS)) {
      int i = floor(nu / (globals::NPHIXSNUINCREMENT * nu_edge)) - 10;
      sigma_bf = photoion_xs[i];
    } else {
      /// use a parameterization of sigma_bf by the Kramers formula
      /// which anchor point should we take ??? the cross-section at the edge or at the highest grid point ???
      /// so far the highest grid point, otherwise the cross-section is not continuous
      sigma_bf = photoion_xs[globals::NPHIXSPOINTS - 1] *
                 pow(nu_edge * (1 + globals::NPHIXSNUINCREMENT * globals::NPHIXSPOINTS) / nu, 3);
    }
    return sigma_bf;
  }

Does this condition then not make sense if there are two phixsdata files being used? If what is done here is dependant on which phixsdata file is being used how would we then apply this condition?

Reply:
I declared phixs_file_version_exists in atomic.h like so:

extern __managed__ std::array<bool, 3> phixs_file_version_exists;

This is where phixs_file_version was previously declared and the atomic.h header is included in the files that use phixs_file_version_exists but I'm getting this error when compiling: 
 
undef: _phixs_file_version_exists
Undefined symbols for architecture x86_64:
  "_phixs_file_version_exists", referenced from:
      _main in lto.o
      photoionization_crosssection_fromtable(float const*, double, double) in lto.o
      read_atomicdata_files() in lto.o
      read_phixs_data(int) in lto.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [sn3d] Error 1

Reply:
I'm defining the boolean array elements like this incase that's relevant:

  phixs_file_version_exists[1] = std::ifstream(phixsdata_filenames[1]).good();
  phixs_file_version_exists[2] = std::ifstream(phixsdata_filenames[2]).good();

Reply:
Those are two assignments, not a definition. You need a definition that matches your declaration, so put 'std::array<bool, 3> phixs_file_version_exists;' in the cc file outside of any functions.
Ignore the managed keyword for now. It's for CUDA support but I'll be removing that in the next release.

Reply:
That worked thanks! Just testing that the code is behaving as expected for the different file configurations then I will update the pull request this afternoon.